PHP/Toolset.md

```diff
@@ -8,45 +8,69 @@ Dependency management with Composer
 * Ways to autoload dependencies in Composer; 
 * composer.json vs composer.lock
 
+Live templates in IDE
+
+Error handling with [Whoops](https://github.com/filp/whoops)
 
 [Carbon](https://github.com/briannesbitt/Carbon): a simple PHP API extension for DateTime 
 
 Debugging app with [Kint](https://github.com/kint-php/kint)
 
 Editorconfig as a way to share common settings between diffeerent IDEs and editors
 
-# Middle Level
 Generation of universally unique identifiers with [uuid library](https://github.com/ramsey/uuid)
 
+Experience with one of the following template engines: Twig, Plates, Aura View etc
+
+# Middle Level
 Code complexity check with [PHPMD](https://github.com/phpmd/phpmd)
 
-Code debug with xDebug
+Code debug with xDebug. Simple and conditional breakpoints.
+
+Code profiling with xDebug 
+
+Code profiling with [XHProf](https://tideways.com/profiler/xhprof-for-php7)
 
 Fake data generation with [Faker](https://github.com/fzaninotto/Faker)
 
 PSR-7 HTTP message interfaces: [Zend Diactoros](https://github.com/zendframework/zend-diactoros)
 
-Experience with one of the following template engines: Twig, Plates, Aura View etc
-
-Error handling with [Whoops](https://github.com/filp/whoops)
+PSR-15: HTTP Server Request Handlers
```

Comment:
Are there libs or tools to use?  Or shall we move this into OOP&OOD.md file?

Reply:
It's already added to OOP&OOD. I'll look at some good implementation to add. But for PSR-15 I do not know yet something so widely used as Zend Diactoros for PSR-7

Reply:
Added and implementations collection. And also changed Zend to Laminas